src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -16,25 +15,20 @@ public sealed class Sagas : Feature
     /// </summary>
     public Sagas()
     {
-        Defaults(s =>
-        {
-            conventions = s.Get<Conventions>();
-
-            var sagas = s.GetAvailableTypes().Where(IsSagaType).ToList();
-            if (sagas.Count > 0)
-            {
-                conventions.AddSystemMessagesConventions(t => IsTypeATimeoutHandledByAnySaga(t, sagas));
-            }
-        });
+        Defaults(s => s.SetDefault(new SagaMetadataCollection()));
 
         Enable<SynchronizedStorage>();
 
-        Defaults(s => s.Set(new SagaMetadataCollection()));
+        DependsOn<SynchronizedStorage>();
 
         Prerequisite(context => !context.Settings.GetOrDefault<bool>("Endpoint.SendOnly"), "Sagas are only relevant for endpoints receiving messages.");
-        Prerequisite(config => config.Settings.GetAvailableTypes().Any(IsSagaType), "No sagas were found in the scanned types");
-
-        DependsOn<SynchronizedStorage>();
+        Prerequisite(context =>
+        {
+            var sagaCollection = context.Settings.Get<SagaMetadataCollection>();
+            var sagaMetadata = SagaMetadata.CreateMany(context.Settings.GetAvailableTypes());
+            sagaCollection.AddRange(sagaMetadata);
+            return sagaCollection.HasMetadata;
+        }, "No sagas were found. Either enable assembly scanning or manually register sagas using AddSaga<TSaga>().");
```

Comment:
Should we tailor the message based on if scanning was on or not? We could check if the available types were not empty?